src/main/java/com/epam/wca/gym/service/impl/ProfileServiceImpl.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.wca.gym.service.impl;
+
+import com.epam.wca.gym.entity.Trainee;
+import com.epam.wca.gym.entity.Trainer;
+import com.epam.wca.gym.service.ProfileService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.security.SecureRandom;
+import java.util.Map;
+@Slf4j
+@Service
+public class ProfileServiceImpl implements ProfileService {
+    @Autowired
+    private Map<String, Integer> usernameCounter;
```

Comment:
Use of field injection is not recommended. Use constructor injection to promote immutability and make dependencies more visible.
```suggestion
    private Map<String, Integer> usernameCounter;
    
    @Autowired
  public TraineeServiceImpl(TraineeDAO traineeDAO) {
      this.traineeDAO = traineeDAO;
  }
```

Reply:
It was written in document to use @Autowired  fo dependency injection into services